node/pkg/processor/vaa_processor.go

```diff
@@ -0,0 +1,473 @@
+	obsvReqSendC chan<- *gossipv1.ObservationRequest
+	// signedInC is a channel of inbound signed VAA observations from p2p
+	signedInC <-chan *gossipv1.SignedVAAWithQuorum
+	// gossipSendC is a channel of outbound messages to broadcast on p2p
+	gossipSendC chan<- []byte
+
+	attestationEvents *reporter.AttestationEventReporter
+	logger            *zap.Logger
+	db                *db.Database
+
+	manager *reactor.Manager[*vaaObservation]
+
+	governor        *governor.ChainGovernor
+	acct            *accountant.Accountant
+	acctReadC       <-chan *common.MessagePublication
+	pythnetVaas     map[string]PythNetVaaEntry
+	pythnetVAAsLock sync.Mutex
+}
+
+// NewVAAReactor creates a new VAAReactor.
+func NewVAAReactor(
+	db *db.Database,
+	msgC <-chan *common.MessagePublication,
+	setC <-chan *common.GuardianSet,
+	gossipSendC chan<- []byte,
+	obsvC <-chan *gossipv1.SignedObservation,
+	obsvReqSendC chan<- *gossipv1.ObservationRequest,
+	signedInC <-chan *gossipv1.SignedVAAWithQuorum,
+	gk *ecdsa.PrivateKey,
+	gst *common.GuardianSetState,
+	attestationEvents *reporter.AttestationEventReporter,
+	g *governor.ChainGovernor,
+	acct *accountant.Accountant,
+	acctReadC <-chan *common.MessagePublication,
+) *VAAReactor {
+	// obsC is an internal buffer that is used to forward pre-filtered messages to the reactor manager.
+	// See VAAReactor.Run for filtering logic.
+	obsC := make(chan *vaaObservation, 10)
+
+	r := &VAAReactor{
+		obsvReqSendC:      obsvReqSendC,
+		signedInC:         signedInC,
+		msgC:              msgC,
+		obsC:              obsC,
+		gossipSendC:       gossipSendC,
+		db:                db,
+		attestationEvents: attestationEvents,
+		governor:          g,
+		acct:              acct,
+		acctReadC:         acctReadC,
+		pythnetVaas:       map[string]PythNetVaaEntry{},
+	}
+
+	manager := reactor.NewManager[*vaaObservation]("vaa", obsC, obsvC, setC, gst, reactor.Config{
+		RetransmitFrequency: 5 * time.Minute,
+		QuorumGracePeriod:   2 * time.Minute,
+		QuorumTimeout:       24 * time.Hour,
+		UnobservedTimeout:   24 * time.Hour,
+		Signer:              reactor.NewEcdsaKeySigner(gk),
+		NetworkAdapter:      reactor.NewChannelNetworkAdapter(gossipSendC),
+	}, r, r)
+	r.manager = manager
+
+	return r
+}
+
+func (p *VAAReactor) Run(ctx context.Context) error {
+	p.logger = supervisor.Logger(ctx)
+
+	err := supervisor.Run(ctx, "vaa-reactor-manager", p.manager.Run)
+	if err != nil {
+		return fmt.Errorf("failed to start reactor manager: %w", err)
+	}
+
+	cleanup := time.NewTicker(30 * time.Second)
+	defer cleanup.Stop()
+
+	reobservationTicker := time.NewTicker(5 * time.Minute)
+	defer reobservationTicker.Stop()
+
+	// Always initialize the timer so don't have a nil pointer in the case below. It won't get rearmed after that.
+	govTimer := time.NewTicker(time.Minute)
+	defer govTimer.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			if p.acct != nil {
+				p.acct.Close()
+			}
+			return ctx.Err()
+		case k := <-p.msgC:
```

Comment:
idea: put token bridge messages (or anything that's covered by governor or accountant) into its own channel and process concurrently; drop them if said channel is full to not hold up processing of other messages.